tests/cases_homogenize.py

```diff
@@ -278,3 +278,62 @@ def case_rasterize_vector_success(
 )
 def case_rasterize_vector_fail(candidate_map, benchmark_map, rasterize_attributes):
     return candidate_map, benchmark_map, rasterize_attributes
+
+
+candidate_int32 = _load_xarray("candidate_map_0.tif")
+candidate_float64 = _load_xarray("candidate_map_1.tif")
```

Comment:
Any issues expected with `np.nan` if `mask_and_scale=True`? Or is this irrelevant?

Reply:
When mask_and_scale=True is applied the datatype is elevated to a float64 or maintains float32 since nans can only be represented in float space.